src/keepass_fmt_plug.c

```diff
@@ -37,115 +42,240 @@ john_register_one(&fmt_KeePass);
 #include "formats.h"
 #include "params.h"
 #include "options.h"
-#include "keepass_common.h"
 #include "sha2.h"
 #include "aes.h"
 #include "twofish.h"
 #include "chacha.h"
+#include "hmac_sha.h"
+#include "argon2.h"
+
+#define KEEPASS_AES                     1
+#define KEEPASS_ARGON2                  1
+#define KEEPASS_REAL_COST_TEST_VECTORS  0
+#include "keepass_common.h"
 
 #ifndef OMP_SCALE
-#define OMP_SCALE               4 // This and MKPC tuned for core i7
+#define OMP_SCALE               1
 #endif
 
 #define FORMAT_LABEL            "KeePass"
 #define FORMAT_NAME             ""
-#define ALGORITHM_NAME          "SHA256 AES 32/" ARCH_BITS_STR
+#if !defined (JOHN_NO_SIMD) && defined(__AVX512F__)
+#define ALGORITHM_NAME          "AES/Argon2 512/512 AVX512F"
+#elif !defined (JOHN_NO_SIMD) && defined(__AVX2__)
+#define ALGORITHM_NAME          "AES/Argon2 256/256 AVX2"
+#elif !defined (JOHN_NO_SIMD) && defined(__XOP__)
+#define ALGORITHM_NAME          "AES/Argon2 128/128 XOP"
+#elif !defined (JOHN_NO_SIMD) && defined(__SSE2__)
+#define ALGORITHM_NAME          "AES/Argon2 128/128 SSE2"
+#else
+#define ALGORITHM_NAME          "AES/Argon2 32/" ARCH_BITS_STR
+#endif
+
+struct argon2_memory {
+	int used;
+	region_t region;
+};
```

Comment:
In the Argon2 format, I've since revised this struct to:

```c
struct argon2_memory {
        region_t region;
        int used;
        char padding[MEM_ALIGN_CACHE - sizeof(region_t) - sizeof(int)];
};
```

Reply:
Oh right, I saw that but forgot.